### ISSUE TITLE
Collecting Output Format for new Java API (only for local executor)

### ISSUE THREAD

**Issue by rmetzger** [stratosphere/stratosphere#602]
``` java
List<String> resultsCollected = new ArrayList<String>();
result.output(resultsCollected);
```

**Comment by qmlmoon** [comment]
When dop = 2, then we'll get 2 collection outputs. How to merge them? They have no idea about how the record fields are grouped. So, do we just simply concat them or leave them like  `List<List<String>> resultCollected` ?

**Comment by fhueske** [comment]
Good observation.
In most cases, you can simply concatenate them since the order of elements is not important.
If the user specifies sorted output, going for nested lists would be the best thing to do.

For now, I would go with the concatenated lists as it is easier to handle for the user.

**Comment by rmetzger** [comment]
This is related: https://github.com/stratosphere/stratosphere/issues/478 (for an implementation into the runtime)

**Comment by fhueske** [comment]
Issue solved in b032ff3afb10f9e0158e133a8f5c447ad89a90b6
Thanks!
